diff --git a/src/amd/terascale/vulkan/terakan_instance.h b/src/amd/terascale/vulkan/terakan_instance.h
index 78db85d7929..d8fd782da55 100644
--- a/src/amd/terascale/vulkan/terakan_instance.h
+++ b/src/amd/terascale/vulkan/terakan_instance.h
@@ -38,7 +38,7 @@ extern "C" {
 #define TERAKAN_USE_WSI_PLATFORM
 #endif
 
-#define TERAKAN_API_VERSION VK_MAKE_API_VERSION(0, 1, 0, VK_HEADER_VERSION)
+#define TERAKAN_API_VERSION VK_MAKE_API_VERSION(0, 1, 1, VK_HEADER_VERSION)
 
 enum {
    TERAKAN_TEST_SHIFT_NO_ALPHA_TO_COVERAGE_DITHERING,
