diff --git a/src/util/cnd_monotonic.c b/src/util/cnd_monotonic.c
index 71d9ac0d247..f0fd455e168 100644
--- a/src/util/cnd_monotonic.c
+++ b/src/util/cnd_monotonic.c
@@ -130,9 +130,9 @@ u_cnd_monotonic_timedwait(struct u_cnd_monotonic *cond, mtx_t *mtx,
    timespec_get(&now_time, TIME_MONOTONIC);
    struct timespec rel_time;
    timespec_sub_saturate(&rel_time, abs_time, &now_time);
-   int rt = pthread_cond_timedwait_relative_np(&cond->cond, mtx, &rel_time);
+   int rt = pthread_cond_timedwait_relative_np(&cond->cond, (pthread_mutex_t *)mtx, &rel_time);
 #else
-   int rt = pthread_cond_timedwait(&cond->cond, mtx, abs_time);
+   int rt = pthread_cond_timedwait(&cond->cond, (pthread_mutex_t *)mtx, abs_time);
 #endif
    if (rt == ETIMEDOUT)
       return thrd_timedout;
@@ -151,6 +151,6 @@ u_cnd_monotonic_wait(struct u_cnd_monotonic *cond, mtx_t *mtx)
                             (PCRITICAL_SECTION)mtx, INFINITE);
    return thrd_success;
 #else
-   return (pthread_cond_wait(&cond->cond, mtx) == 0) ? thrd_success : thrd_error;
+   return (pthread_cond_wait(&cond->cond, (pthread_mutex_t *)mtx) == 0) ? thrd_success : thrd_error;
 #endif
 }
